rfcs/RFC-Upload-From-Dropbox.md

```diff
@@ -0,0 +1,417 @@
+
+**Response:**
+
+| Key        | Description                                                               |
+| ---------- | ------------------------------------------------------------------------- |
+| dataset_id | identifies the dataset the file will be associated with after validation. |
+| status     | Provides the current status of the upload                                 |
+
+**Error Responses:**
+
+| Code | Description                                                                               |
+| ---- | ----------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission |
+| 400  | if the file type is invalid.                                                              |
+| 409  | if there is an existing submissions in progress with the same dataset_id.                 |
+| 413  | if the links refer to a file that exceeds the max size allowed.                           |
+
+#### GET submission/{submission_id}/upload
+
+Checks the status of an existing upload job.
+
+**Request:**
+
+| Parameter     | Description                          |
+| ------------- | ------------------------------------ |
+| submission_id | identifies the submission            |
+| dataset_id    | identifies the dataset to be queried |
+
+**Response:**
+
+| Key      | Description                                   |
+| -------- | --------------------------------------------- |
+| status   | Provides the current status of the upload.    |
+| message  | If an error occurred, the message shows here. |
+| progress | The current progress of the upload.           |
+
+**Error Responses:**
+
+| Code | Description                                                                                                             |
+| ---- | ----------------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress upload. |
+| 400  | if the parameters supplied are invalid.                                                                                 |
+
+#### DELETE submission/{submission_id}/upload
+
+Cancels an existing upload job. Any data that has started to upload is removed, and the upload status is changed to
+cancel pending until the job has been cleared up.
+
+**Request:**
+
+| Parameter     | Description                           |
+| ------------- | ------------------------------------- |
+| submission_id | identifies the submission             |
+| dataset_id    | identifies the dataset being canceled |
+
+**Response:**
+
+| Key    | Description                                |
+| ------ | ------------------------------------------ |
+| status | Provides the current status of the upload. |
```

Comment:
If the status is always `Cancel Pending`, then there's no reason to return status.

What is the success return code? Is it `202 Accepted` since this is asynchronous?

Reply:
I added return codes 202 for all of the Async requests

Reply:
Remove the extra response parameter